app/src/main/java/org/mozilla/fenix/compose/cfr/CFRPopup.kt

```diff
@@ -72,6 +72,19 @@ class CFRPopup(
      */
     fun show() {
         anchor.post {
+            // When we're in this Runnable, the 'show' method might have been called right before
+            // the activity is no longer attached to the WindowManager. When we get to calling
+            // the CFRPopupFullscreenLayout#show method below, we are now trying to attach the View
+            // with the WindowManager that has an unusable Activity.
+            //
+            // To protect against this, within this same Runnable, we check if the anchor view is
+            // safe to use before continuing.
+            //
+            // See: https://bugzilla.mozilla.org/show_bug.cgi?id=1799996
+            if (anchor.context == null && !anchor.isAttachedToWindow) {
```

Comment:
The context would probably always be valid here.
Maybe we can use an `or` check instead?

Testing with manually detaching the anchor and a small delay I'm getting another exception - `java.lang.IllegalStateException: ViewTreeLifecycleOwner not found from org.mozilla.fenix.compose.cfr.CFRPopupFullscreenLayout`.

We could also just check [this token](https://github.com/mozilla-mobile/fenix/blob/8ed3bf704f8d8b9b2e3b9142c994a08c2ad33cce/app/src/main/java/org/mozilla/fenix/compose/cfr/CFRPopupFullscreenLayout.kt#L354) for null and abort if wanting to focus on the `BadTokenException` but checking if the anchor is attached earlier in code seems better - better visibility, avoid extra calls until the abort.

Can the `show` method be updated to return a Boolean so that callers will know when a call to show a CFR failed and try it again later / only persist in shared prefs that the CFR was shown only if this method returned `true`?

Reply:
> Testing with manually detaching the anchor and a small delay I'm getting another exception - java.lang.IllegalStateException: ViewTreeLifecycleOwner not found from org.mozilla.fenix.compose.cfr.CFRPopupFullscreenLayout.

Yeah, it was quite hard to test this manually. With some suggestions from Christian, we tried:
- Adding a delay before adding the view (so that we could leave the fragment).
- Removing the anchor, delaying, adding the view.
- Launching another Activity in the Runnable before adding the view.

In all those cases we weren't able to reproduce the same stack trace. 🥲 

> The context would probably always be valid here.
> Maybe we can use a `logical OR - ||` check instead?

That was my suspicion as well about the context. I figured the `isAttachedToWindow` is enough, but alone it could have made an assumption that the WindowManager's context is still accessible. This is speculative, so I was trying to be overcautious. 😄 The logical OR sounds better for that reason.